modules/twinklebatchdelete.js

```diff
@@ -571,20 +573,34 @@ Twinkle.batchdelete.callbacks = {
 	},
 	deleteRedirectsMain: function(apiobj) {
 		var response = apiobj.getResponse();
-		var pages = response.query.pages[0].redirects || [];
-		pages = pages.map(function(redirect) {
-			return redirect.title;
+		var pages = response.query.pages || [];
+
+		// build list of redirects and their talk pages to delete
+		var pagesToDelete = [];
+		$.each(pages, function(i, page) {
+			// build list of redirects
+			pagesToDelete.push(page.title);
+
+			// build list of redirect talk pages, if needed
+			var isRedirectOnATalkPage = mw.Title.isTalkNamespace(page.ns);
+			var hasExistingTalkPage = typeof page.talkid !== 'undefined';
+			if (!isRedirectOnATalkPage && hasExistingTalkPage) {
+				var talkPageTitle = new mw.Title(page.title).getTalkPage().toText();
+				pagesToDelete.push(talkPageTitle);
+			}
 		});
-		if (!pages.length) {
+
+		if (!pagesToDelete.length) {
 			return;
 		}
 
-		var redirectDeleter = new Morebits.batchOperation('Deleting redirects to ' + apiobj.params.page);
+		// delete redirects and redirect talk pages
+		var redirectDeleter = new Morebits.batchOperation('Deleting redirects to ' + apiobj.params.page + ' and their associated talk pages');
 		redirectDeleter.setOption('chunkSize', Twinkle.getPref('batchChunks'));
-		redirectDeleter.setPageList(pages);
+		redirectDeleter.setPageList(pagesToDelete);
 		redirectDeleter.run(function(pageName) {
 			var wikipedia_page = new Morebits.wiki.page(pageName, 'Deleting ' + pageName);
-			wikipedia_page.setEditSummary('[[WP:CSD#G8|G8]]: Redirect to deleted page "' + apiobj.params.page + '"');
+			wikipedia_page.setEditSummary('[[WP:CSD#G8|G8]]: Redirect or redirect talk page to deleted page "' + apiobj.params.page + '"');
```

Comment:
I think it would be better to do two batch operations as you had in an earlier iteration so distinct log summaries are used for the two cases – "Redirect to deleted page" or "Talk page of redirect".